test/date_time_parser_test.exs

```diff
@@ -185,6 +185,27 @@ defmodule DateTimeParserTest do
     )
 
     test_datetime_parsing("0000000000", DateTime.from_naive!(~N[1970-01-01T00:00:00], "Etc/UTC"))
+
+    test_datetime_parsing("-0000000001", DateTime.from_naive!(~N[1969-12-31T23:59:59], "Etc/UTC"))
+
+    test_datetime_parsing(
+      "-0000000001.0000000001",
+      DateTime.from_naive!(~N[1969-12-31T23:59:58.000000], "Etc/UTC")
+    )
+
+    # example from the Wikipedia article
```

Comment:
**This may end up warranting a full discussion in the original thread, but I'll start it here**

From the [Wikipedia article](https://en.wikipedia.org/wiki/Unix_time):

> This can be extended backwards from the epoch too, using negative numbers; thus 1957-10-04T00:00:00Z, 4472 days before the epoch, is represented by the Unix time number −4472 × 86400 = −386380800

In this example, the final number, `-386380800` is only nine digits. 

Currently, the `epoch_regex` expect seconds to have ten or eleven digits:

https://github.com/taxjar/date_time_parser/blob/42a93ff734bc532f86ca3a0bf90b662ce3bfade8/lib/date_time_parser.ex#L106

As a result, this test will fail:

```elixir
test_datetime_parsing("-386380800", DateTime.from_naive!(~N[1957-10-04T00:00:00], "Etc/UTC"))
```

adding a leadig zero will make it pass:

```elixir
test_datetime_parsing("-0386380800", DateTime.from_naive!(~N[1957-10-04T00:00:00], "Etc/UTC"))
```

Based on the example you gave in [the original issue](https://github.com/taxjar/date_time_parser/issues/10), it seems like Ruby may not care about the number of digits either. 

For now, I am leaving things as-is an adding the leading zero. I think if we want to support fewer digits with negative numbers, it'd probably be good to make a new issue.